File: examples/metal/device/example_1/go/main.go
package main

import (
	"github.com/equinix/pulumi-equinix/sdk/go/equinix/metal"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := metal.NewDevice(ctx, "web1", &metal.DeviceArgs{
			Hostname:        pulumi.String("tf.coreos2"),
			Plan:            pulumi.String(metal.PlanC3SmallX86),
			Metro:           pulumi.String("sv"),
			OperatingSystem: pulumi.String("ubuntu_24_04"),
			BillingCycle:    pulumi.String(metal.BillingCycleHourly),
			ProjectId:       pulumi.Any(projectId),
		})
		if err != nil {
			return err
		}
		return nil
	})
}

File: examples/metal/device/example_1/python/__main__.py
import pulumi
import pulumi_equinix as equinix

web1 = equinix.metal.Device("web1",
    hostname="tf.coreos2",
    plan=equinix.metal.Plan.C3_SMALL_X86,
    metro="sv",
    operating_system="ubuntu_24_04",
    billing_cycle=equinix.metal.BillingCycle.HOURLY,
    project_id=project_id)
